### PR TITLE
feat: Page on writing insights

### PR DESCRIPTION
**Pull request by Tinystep1**
Added page on code needed when writing insights in order for it to work correctly.

refs: 10167

### CODE REVIEW

File: _documentation/core-modules/insight/insight.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+class ExampleInsight extends InsightBaseClass
+```
+
+Below this, give the variables for the insights name and description. These will appear in the insight list, the first page seen in cmfive when you click on the insight module.
+
+```php
+public $name = "Example Insight";
+public $description = "This insight is being used to show correct insight syntax in the docs";
+```
+These variables will indicate to users which isnight is which.
```
Comment (strongishllama) [inline _documentation/core-modules/insight/insight.md]: @Tinystep1 Small typo on ```insight``` here.

File: _documentation/core-modules/insight/insight.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+All insights will contain a getFilters function and a run function. The getFilters function will be called when a user clicks on the View button next to the insight. The run function is called when the user clicks Run from the view screen.
+
+getFilters will be an array, which sets up the parameters the user can choose. Before viewing an insight, the user may specify it to only show information from specififc dates, for certain users, etc.<br>
```
Comment (strongishllama) [inline _documentation/core-modules/insight/insight.md]: @Tinystep1 Small typo on ```specific``` here.

**Review by strongishllama — CHANGES_REQUESTED** [review]
Hey, @Tinystep1 nice work. Just two small typos to fix. I've been using this VS Code [extension](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) to catch my typos, thought you might find it helpful!

**Review by strongishllama — APPROVED** [review]

**Comment by Tinystep1** [comment]
Thanks. I had been copying it from my localhost and running it through
Grammarly, so I'm surprised I didn't pick up on these. I'll try the
extension too

**Pull request merged**